### PR TITLE
cmd/geth: check block range against chain head in export cmd

### LINKED ISSUE #22383: Export cmd should fail if chain length is shorter than input range
#### System information

Geth version: `1.9.24-unstable`
OS & Version:Linux
Commit hash : `0391daf9abef68e5da078a24f7d214270dbb484d`

#### Expected behaviour

After fetching the tip of the chain from db, `export` should compare that against the block range provided by user and fail if range exceeds the local chain.

#### Actual behaviour

It starts exporting block batches and fails once it reaches the tip of local chain with `Fatal: Export error: export failed on #3863673: not found`.

#### Steps to reproduce the behaviour

For e.g. `geth export chain.gz 0 15000000`

### PR DESCRIPTION
**Pull request by s1na**
Fixes #22383

### CODE REVIEW

File: cmd/geth/chaincmd.go [hunk 1]
```diff
@@ -341,6 +341,10 @@ func exportChain(ctx *cli.Context) error {
 		if first < 0 || last < 0 {
 			utils.Fatalf("Export error: block number must be greater than 0\n")
 		}
+		head := chain.CurrentBlock()
+		if uint64(last) > head.NumberU64() {
```
Comment (holiman) [inline cmd/geth/chaincmd.go:345]: More go-idiomatic form:
```suggestion
		if head := chain.CurrentBlock(); uint64(last) > head.NumberU64() {
```

**Review by holiman — APPROVED** [review]
LGTM

File: cmd/geth/chaincmd.go [hunk 2]
```diff
@@ -341,6 +341,9 @@ func exportChain(ctx *cli.Context) error {
 		if first < 0 || last < 0 {
 			utils.Fatalf("Export error: block number must be greater than 0\n")
 		}
+		if head := chain.CurrentBlock(); uint64(last) > head.NumberU64() {
```
Comment (karalabe) [inline cmd/geth/chaincmd.go:344]: If you were to do CurrentFastBlock, then export would work on a not-yet-synced node too like now.

Comment (holiman) [reply cmd/geth/chaincmd.go:344]: Yup, ping on this -- the exporter doesn't need the full CurrentBlock, it's sufficient with the CurrentFastBlock, which has the full body of the block.

Comment (s1na) [reply cmd/geth/chaincmd.go:344]: Ah yes that's better. Updated to use `CurrentFastBlock`. If the PR is okay otherwise I can squash the commits

**Pull request merged**